velociraptor/autoplotter/objects.py

```diff
@@ -754,61 +756,51 @@ def get_quantity_from_catalogue_with_mask(
         # in versions of unyt less than 2.6.0
         name = x.name
 
+        # temporary masked array to apply global mask
+        # in concert with plot-specific masks
+        x_mask = self.global_mask
+        
         if self.structure_mask is not None:
-            x = x[self.structure_mask]
+            # if structure_mask already set, mask and return
+            x_mask = logical_and(x_mask, self.structure_mask)
+            x = x[x_mask]
             x.name = name
-        elif self.selection_mask is not None:
+            return x
```

Comment:
Ah I see - in the master version, all the code below this if statement (provided the program enters it) won't be executed because the other branches start with `elif`. 

But you turned all those `elif`s into `if`s, and in order to not let the program read the lines below you have this additional `return` in the middle of the code.

Maybe it is best to come back to the original `if `and `elif` structure to improve the readability of the code? I personally find it a bit confusing when I see unexpected return statements in the middle of a function. What do you think?

Reply:
@EvgeniiChaikin Yeah, I understand there are drawbacks with this approach too. The reason I implemented this way was that following the previous implementation means that the new global mask adds another layer of nesting to the `if`s / `elif`s. I thought this was too convoluted, and in fact this method is clearer to me than nesting anyway (though I may be alone in that, and depends on the comments making it clear _*why*_ we also can return at that point)

Reply:
@jtrayford thanks for explaining!  Yes, I see that having this `return` may shorten the code.

Perhaps, we could ask @JBorrow to add one more opinion on whether this `return` looks OK?

Reply:
Yeah happy to exit early in conditionals like this, as long as the logic remains relatively simple.

Reply:
Good, then let's leave it as is